Templates/TypeScript/src/entity_types.ts.tt

```diff
@@ -9,45 +10,159 @@
     var entityTypes             = model.GetEntityTypes();
     var enumTypes               = model.GetEnumTypes();
     var complexTypes            = model.GetComplexTypes();
+
+    var maxLineLength = 180;
 #>
-// Type definitions for the Microsoft Graph <VERSION_STRING>
+// Type definitions for non-npm package microsoft-graph <VERSION_STRING>
 // Project: https://github.com/microsoftgraph/msgraph-typescript-typings
 // Definitions by: Microsoft Graph Team <https://github.com/microsoftgraph>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 // TypeScript Version: 2.1
 
 export as namespace microsoftgraph;
 
-<# foreach(var enumType in enumTypes) { #>
-export type <#= enumType.Name.UpperCaseFirstChar() #> = <#= enumType.GetEnumValues() #>;
-<# } #>
 <#
-  foreach(var entityType in entityTypes)
-    {
-        var methods = entityType.Methods;
+    foreach(var enumType in enumTypes) {
+        var enumTypeName = enumType.Name.UpperCaseFirstChar();
+        var enumValues = enumType.GetEnumValues();
+        var exportTypeLength = "export type".Length + enumTypeName.Length + enumValues.Length + 3;
+        if (exportTypeLength < maxLineLength) {
+#>
+export type <#= enumTypeName #> = <#= enumValues #>;
+<#
+        } else {
+#>
+export type <#= enumTypeName #> =<#
+            var enums = enumValues.Split('|');
+            for(var i = 0; i < enums.Length; i++) {
+#>
+
+    | <#= enums[i].Trim() #><#
+            }
+#>;
+<#
+        }
+    }
+
+    foreach(var entityType in entityTypes) {
+        var propCount = entityType.Properties.ToList().Count;
+        var entityTypeName = entityType.Name.UpperCaseFirstChar();
+        if(propCount == 0 && entityTypeName[0] == 'I') {
+#>
+// tslint:disable-next-line: interface-name no-empty-interface
+<#
+        } else if (entityTypeName[0] == 'I') {
```

Comment:
`else if (entityTypeName[0] == 'I')` should be `else if (entityTypeName[0] != 'I')` as we want to disable tslint when the interface does not start with 'I'. This doesn't account for types that are named with 'I'. I think we should disable this check on all interfaces since we don't prefix interface names with 'I'

Reply:
Actually definitely typed repo is not allowing the interfaces starting with 'I'.

Reply:
Got it.